src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -274,3 +274,21 @@ def get_storage_blob_uri(cli_ctx, storage):
             raise CLIError('{} does\'t exist.'.format(storage))
         storage_uri = storage_account.primary_endpoints.blob
     return storage_uri
+
+
+def vm_secret_format_output_handler(ctx, **kwargs):
+    args = kwargs.get("args", None)
+    if args and " ".join(args[0:3]) == "vm secret format":
+        allowed_formats = ctx.output_cls._FORMAT_DICT.keys()
+        output_options = ["--output", "-o", "--out"]
+        desired_formats = ["json", "jsonc"] # todo: add yaml?
```

Comment:
Okay, I agree.  I had considered adding yaml because it is one of the few formats that has meaningful output.  Whereas the other commands tsv / table essentially don't have any output.

Reply:
You would then have to make the other command (which I believe is`vm create`) accept YAML format as well. Not in any way worth the hassle.